NWN.Anvil/src/main/Services/API/Creature/InitiativeModifierService.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using Anvil.API;
+using Anvil.Internal;
+using NWN.Native.API;
+using Feat = Anvil.API.Feat;
+
+namespace Anvil.Services
+{
+  [ServiceBinding(typeof(InitiativeModifierService))]
+  [ServiceBindingOptions(InternalBindingPriority.API, Lazy = true)]
+  public sealed unsafe class InitiativeModifierService
+  {
+    private readonly FunctionHook<InitiativeModifier> initiativeModiferHook;
+
+    public InitiativeModifierService(HookService hookService)
+    {
+      initiativeModiferHook = hookService.RequestHook<InitiativeModifier>(OnResolveInitiative, FunctionsLinux._ZN12CNWSCreature17ResolveInitiativeEv, HookOrder.Late);
+    }
+
+    private delegate void InitiativeModifier(void* pObject);
+
+    /// <summary>
+    /// Clears any modifier that is set for the creature's initiative.<br/>
+    /// </summary>
+    public void ClearInitiativeModifier(NwCreature creature)
+    {
+      InternalVariables.InitiativeMod(creature).Delete();
+    }
+
+    /// <summary>
+    /// Gets the modifier that is set for the creature's initiative.<br/>
+    /// </summary>
+    public int? GetInitiativeModifier(NwCreature creature)
+    {
+      InternalVariableInt initiativeModifier = InternalVariables.InitiativeMod(creature);
+      if (initiativeModifier.HasValue)
+      {
+        return InternalVariables.InitiativeMod(creature);
```

Comment:
Can reuse the variable here:

```suggestion
        return initiativeModifier.Value;
```